src/routes/PlayQuiz/index.tsx

```diff
@@ -1,71 +1,111 @@
+  const { timeLimit, selectionTimeLimit, isQuestionPointsHidden } = gameInfo;
   const [isPlaying, setIsPlaying] = useState(false);
+  const [isGameStarted, setIsGameStarted] = useState(false);
   const [winner, setWinner] = useState('');
-  const selectedOptionsData = gameInfo.teams.reduce(
-    (acc, team) => acc.concat(team.selectedOptions),
+  const selectedOptionsData: any = gameInfo.teams.reduce(
+    (acc, team) => acc.concat(team.selectedOptions || []),
     [] as SelectedOptions[],
   );
   const attemptedQuestionIds = selectedOptionsData.map((x) => x.questionId);
   const isQuestionAttempted = !!selectedQuestion && attemptedQuestionIds.includes(selectedQuestion.questionId);
   const showQuestionTimer = !!timeLimit && !!selectedQuestion && !isQuestionAttempted;
   const { showModal, getGameData, updateGame } = useStore();
   const [isLoading, setIsLoading] = useState(true);
-  const navigate = useNavigate();
 
   useEffect(() => {
     if (gameId) {
       getGameData(parseInt(gameId)).then((game) => {
+        debugger;
         const { quiz, ...restData } = game;
         const quizData = quiz[0];
-        const categoryIds = quizData.categories.map((category) => category.categoryId);
-        setQuizInfo({
-          quizId: quizData.quizId,
-          name: quizData.name,
-          categoryIds,
-          numberOfQuestionsPerCategory: quizData.numberOfQuestionsPerCategory,
+        let questionNum = 1;
+        let allQuestionsCount = 0;
+
+        quizData.categories.forEach((category) => {
+          category.questions.forEach((q) => {
+            q.questionNum = questionNum;
+            questionNum += 1;
+            allQuestionsCount++;
+          });
         });
 
-        if (categoryIds.length > 0) {
-          setCategoriesInfo(formatCategoryInfo(quizData.categories, categoryIds));
+        setQuizInfo(quizData);
+        setGameInfo(restData);
+        const attemptedQuestionsCount = restData.teams.reduce((count, team) => {
+          count += team.selectedOptions.length;
+          return count;
+        }, 0);
+        const isCompleted = allQuestionsCount === attemptedQuestionsCount;
+
+        if (isCompleted) {
+          const winners = getWinners(restData.teams);
+          const winnerIds = winners.map((winner) => winner.teamId).join(',');
+          setWinner(winnerIds);
+        } else if (attemptedQuestionsCount > 0) {
+          setSelectedQuestion(getLastAttemptedQuestion(restData.teams, quizData.categories));
         }
 
-        setGameInfo(restData);
+        setIsGameStarted(attemptedQuestionsCount > 0);
         setIsLoading(false);
       });
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [gameId]);
 
+  function startGame() {
+    setIsPlaying(!!gameInfo.selectionTimeLimit);
+    setIsGameStarted(true);
+  }
+
+  function getLastAttemptedQuestion(teams: Team[], categories) {
+    const clonedTeams = JSON.parse(JSON.stringify(teams));
```

Comment:
can we use deepClone recursive function instead?

Reply:
deepClone function is useful if we have any `Date` value in the object. for simple nested object `JSON.parse(JSON.stringify())` is 800 times more faster than deepClone function. 

See the second answer for more details
https://stackoverflow.com/questions/122102/what-is-the-most-efficient-way-to-deep-clone-an-object-in-javascript

Reply:
OK